cereus/src/main/java/com/cactus/adapters/WebGroceryAdapter.java

```diff
@@ -2,42 +2,262 @@
+            ArrayList<String> itemNames = new ArrayList<>();
+            for(JsonNode jsonNode1: arrayNode){
+                String itemName = jsonNode1.get("name").asText();
+                itemNames.add(itemName);
+            }
+
+            // Create GroceryItem objects
+            ArrayList<GroceryItem> groceryItems = new ArrayList<>();
+            for (String name: itemNames) {
+                GroceryItem groceryItem = new GroceryItem(name, listID);
+                groceryItems.add(groceryItem);
+            }
+            return groceryItems;
+
+        }catch (NullPointerException | IOException i) {
+            i.printStackTrace();
+            return null;
+        }
+
     }
 
     @Override
-    public GroceryList createGroceryList(String nameList, long userid) {
-        return null;
+    public GroceryList createGroceryList(String nameList, String token) {
+        OkHttpClient client = new OkHttpClient();
+
+        HttpUrl.Builder baseUrl = new HttpUrl.Builder().scheme("http").host("192.168.0.127").port(8080);
+
+        HttpUrl url = baseUrl
+                .addPathSegment("api")
+                .addPathSegment("create-list")
+                .addQueryParameter("name", nameList)
+                .build();
+
+        // Create body
+        RequestBody requestBody = RequestBody.create("",
+                MediaType.get("application/json; charset=utf-8"));
+
+        Request request = new Request.Builder()
+                .url(url)
+                .addHeader("Authorization", token)
+                .post(requestBody)
+                .build();
+        try {
+        // Make request
+        Response response = client.newCall(request).execute();
+
+        // Parse response
+        ObjectMapper finalMapper = new ObjectMapper()
+                .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+        if (response.code() != HTTP_OK) {
+            System.out.println(response.code());
+            return null;
+        }
+            return finalMapper.readValue(Objects.requireNonNull(response.body()).string(), GroceryList.class);
+        }
+        catch(NullPointerException | IOException e){
+            e.printStackTrace();
+            return null;
+        }
     }
 
     @Override
-    public boolean setGroceryItems(List<String> items, long listid, long userid) {
-        return false;
+    public boolean setGroceryItems(List<String> items, long listID, String token) {
+        OkHttpClient client = new OkHttpClient();
+
+        HttpUrl.Builder baseUrl = new HttpUrl.Builder().scheme("http").host(STATIC_IP).port(8080);
+        GroceryList list = getGroceryList(listID, token);
+        HttpUrl url = baseUrl
+                .addPathSegment("api")
+                .addPathSegment("save-list")
+                .build();
+
+        HashMap<String, Object> save = new HashMap<>();
+        save.put("id", listID);
+        save.put("name", list.getName());
```

Comment:
In this method, you are making two requests (one in `getGroceryList`), when this method should only be one. It is not necessary to include the name, even though I know the documentation I provided says you do. Not providing the name will just keep the name unchanged. Since we are not supporting editing the name of a list for now, please remove the extra call.

Reply:
Yeah sorry I just did that cause I was getting an error and trying to find where it is